lua/lsp/null-ls/services.lua

```diff
@@ -0,0 +1,55 @@
+local M = {}
+
+local logger = require("core.log"):get_default()
```

Comment:
This is still not safe :)

Reply:
This one might be safe because it relies on null-ls being loaded (with plenary as dep), but that's not really the issue here.
IMO, the issue is that we return a nil logger when `plenary` is missing, this forces us to have `Log:get_default()` everywhere we want to log, instead of instantiating a module logger, this I really don't like.
Also, because of that, we still mix `print` calls where we know that the dependency could not yet be available.
A quick solution to solve this issue could be to provide a basic logger which would implement the same interface as `planary`'s one and dispatch the calls to `print`, this way we at least keep the same logging abstraction while providing basic capabilities.
Do you mind leaving this as is, expect `plenary` to be installed, and leave it to another PR instead ?

Reply:
Yeah I mentioned this same exact issue yesterday. We need to figure out a way to either have `plenary` as a direct dependency, (e,g. clone it while installing, or as sub-module,.. etc) or have own logger module, that could optionally be 1:1 feature-wise with `plenary.log`.

I'll create a ticket for this, so that we can see the pros and cons of each approach.